src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.cs

```diff
@@ -130,5 +148,64 @@ private void OnErrorListPropertyChanged(object sender, PropertyChangedEventArgs
                 AddTableSourceIfNecessary(this.Workspace.CurrentSolution);
             }
         }
+
+        private void OnErrorListFullSolutionAnalysisToggled(object sender, AnalysisToggleStateChangedEventArgs e)
+        {
+            var newOptions = _optionService.GetOptions()
+                .WithChangedOption(RuntimeOptions.FullSolutionAnalysis, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.CSharp, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.VisualBasic, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, TypeScriptLanguageName, e.NewState);
+            
+            _optionService.SetOptions(newOptions);
+        }
+
+        private void OnOptionChanged(object sender, OptionChangedEventArgs e)
+        {
+            Contract.ThrowIfFalse(_errorList is IErrorList2);
+
+            if (e.Option == RuntimeOptions.FullSolutionAnalysis || e.Option == ServiceFeatureOnOffOptions.ClosedFileDiagnostic)
+            {
+                var analysisDisabled = !(bool)e.Value;
+                if (analysisDisabled)
+                {
+                    ((IErrorList2)_errorList).AnalysisToggleState = false;
+                }
+            }
+        }
+
+        internal static bool ErrorListHasFullSolutionAnalysisButton()
+        {
+            try
+            {
+                // Full solution analysis option has been moved to the error list from Dev14 Update3.
+                // Use reflection to check if the new interface "IErrorList2" exists in Microsoft.VisualStudio.Shell.XX.0.dll.
+                return Assembly.GetAssembly(typeof(ErrorHandler)).GetType("Microsoft.Internal.VisualStudio.Shell.IErrorList2") != null;
```

Comment:
My understanding (that could be very wrong) was that the JIT tried to compile your *entire* method as soon as you entered it, so it would demand IErrorList2 must exist _before_ your code ever gets to the point where it's checking. Generally the pattern has been separate methods get involved, sometimes marked NoInline. But as long as you've tested, :+1:.